### PR TITLE
Add a failing test case for linker pruning

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt [hunk 1]
```diff
@@ -20,6 +20,9 @@ import com.google.common.jimfs.Jimfs
 import com.squareup.wire.StringWireLogger
 import com.squareup.wire.kotlin.RpcCallStyle
 import com.squareup.wire.kotlin.RpcRole
+import com.squareup.wire.testing.add
```
Comment (lkerford) [inline wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt]: I don't think these imports are required

File: wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+
+fun FileSystem.add(pathString: String, contents: String) {
+  val path = getPath(pathString)
```
Comment (lkerford) [inline wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt]: Should we be checking is 'pathString' is empty? Should we be handling invalid paths?

File: wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  if (path.parent != null) {
+    Files.createDirectories(path.parent)
+  }
+  Files.write(path, contents.toByteArray(Charsets.UTF_8))
+}
+
+fun FileSystem.symlink(linkPathString: String, targetPathString: String) {
+  val linkPath = getPath(linkPathString)
```
Comment (lkerford) [inline wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt]: Should we be checking is 'pathString' is empty? Should we be handling invalid paths?

File: wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        result.add(path.toString())
+      }
+      return FileVisitResult.CONTINUE
+    }
+  })
+  return result
+}
+
+fun FileSystem.addZip(pathString: String, vararg contents: Pair<String, String>) {
+  val path = getPath(pathString)
```
Comment (lkerford) [inline wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt]: Should we be checking is 'pathString' is empty? Should we be handling invalid paths?

File: wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt (hunk 1 above)
Comment (swankjesse) [reply wire-compiler/src/test/java/com/squareup/wire/schema/WireRunTest.kt]: I moved files out of this package, so they used to not need imports and now they do!

File: wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt (hunk 2 above)
Comment (swankjesse) [reply wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt]: nah, for test code like this if it fails it fails

File: wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt (hunk 3 above)
Comment (swankjesse) [reply wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt]: I think it’s mostly use-case dependent. This isn’t a public API so we don’t anticipate invalid inputs.

File: wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt (hunk 4 above)
Comment (swankjesse) [reply wire-test-utils/src/main/java/com/squareup/wire/testing/files.kt]: same.

**Pull request merged**